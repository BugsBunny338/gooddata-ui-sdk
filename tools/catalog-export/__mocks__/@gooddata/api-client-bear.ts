// (C) 2007-2018 GoodData Corporation
import catalogItemsResponse from "./loadItems.json";
import getObjectsResponse from "./getObjects.json";
import getObjectsByQueryResponse from "./getObjectsByQuery.json";
import getVisualizationsResponse from "./getVisualizations.json";

export default {
    catalogue: {
        loadItems() {
            return new Promise((resolve) => resolve(catalogItemsResponse));
        },
    },
    md: {
        getObjectsByQuery() {
            return new Promise((resolve) => resolve(getObjectsByQueryResponse));
        },
        getObjects() {
            return new Promise((resolve) => resolve(getObjectsResponse));
        },
        getVisualizations() {
            return new Promise((resolve) => resolve(getVisualizationsResponse));
        },
    },
};
