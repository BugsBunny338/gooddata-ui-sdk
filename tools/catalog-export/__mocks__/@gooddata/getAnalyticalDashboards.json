[
    {
        "author": "/gdc/account/profile/187aa70b3f3ed50ba4940e7b1966367d",
        "category": "analyticalDashboard",
        "contributor": "/gdc/account/profile/2e5a40d590f955c88251a8aead5267bd",
        "created": "2017-01-26 17:53:13",
        "deprecated": "0",
        "identifier": "afMA17GSbk31",
        "isProduction": 1,
        "summary": "",
        "tags": "",
        "title": "KPIs",
        "updated": "2020-11-18 22:14:11",
        "uri": "/gdc/md/mbuumy476p78ybcceiru61hcyr8i8lo8/obj/2663"
    },
    {
        "author": "/gdc/account/profile/187aa70b3f3ed50ba4940e7b1966367d",
        "category": "analyticalDashboard",
        "contributor": "/gdc/account/profile/187aa70b3f3ed50ba4940e7b1966367d",
        "created": "2017-05-03 15:03:50",
        "deprecated": "0",
        "identifier": "abBJlnxrfEWH",
        "isProduction": 1,
        "summary": "",
        "tags": "",
        "title": "Store management KPIs",
        "updated": "2018-01-23 04:43:37",
        "uri": "/gdc/md/mbuumy476p78ybcceiru61hcyr8i8lo8/obj/6853"
    },
    {
        "author": "/gdc/account/profile/187aa70b3f3ed50ba4940e7b1966367d",
        "category": "analyticalDashboard",
        "contributor": "/gdc/account/profile/187aa70b3f3ed50ba4940e7b1966367d",
        "created": "2018-11-12 11:29:01",
        "deprecated": "0",
        "identifier": "aahlrtUEgKkq",
        "isProduction": 1,
        "summary": "",
        "tags": "",
        "title": "Untitled",
        "updated": "2018-11-12 11:29:02",
        "uri": "/gdc/md/mbuumy476p78ybcceiru61hcyr8i8lo8/obj/8867"
    }
]
