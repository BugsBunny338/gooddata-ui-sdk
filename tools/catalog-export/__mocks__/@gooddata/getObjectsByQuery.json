[
    {
        "dataSet": {
            "content": {
                "urn": "urn:gooddata:date",
                "mode": "",
                "facts": [],
                "titleSuffix": "Timeline",
                "dataLoadingColumns": [],
                "attributes": [
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/439",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/440",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/441",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/442",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/443",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/444",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/445",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/446",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/447",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/448",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/449",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/450",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/451",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/452",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/453",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/454",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/455",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/456"
                ],
                "identifierPrefix": "timeline"
            },
            "links": {
                "dataUploads": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/data/uploads/499"
            },
            "meta": {
                "author": "/gdc/account/profile/8d381a33fff78cf82392c837246d362e",
                "uri": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/499",
                "tags": "",
                "created": "2010-08-02 08:13:42",
                "identifier": "timeline.dataset.dt",
                "deprecated": "0",
                "summary": "DataSet Date",
                "isProduction": 1,
                "title": "Date (Timeline)",
                "category": "dataSet",
                "updated": "2018-05-15 16:18:26",
                "contributor": "/gdc/account/profile/9bf17dfc423644e9bf92925bab65a3e9"
            }
        }
    },
    {
        "dataSet": {
            "content": {
                "mode": "SLI",
                "facts": [
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/706",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/707"
                ],
                "customUploadTimestamp": 1566890302945695,
                "dataLoadingColumns": [
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/62",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/64",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/132",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/134",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/17",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/19",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/21",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/23",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/25",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/27",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/29",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/31",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/33",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/35",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/37"
                ],
                "attributes": [
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/679",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/680",
                    "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/682"
                ]
            },
            "links": {
                "dataUploads": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/data/uploads/165"
            },
            "meta": {
                "author": "/gdc/account/profile/8d381a33fff78cf82392c837246d362e",
                "uri": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/165",
                "tags": "",
                "created": "2018-04-27 11:13:04",
                "identifier": "dataset.salesdetailfact",
                "deprecated": "0",
                "summary": "",
                "isProduction": 1,
                "title": "Sales Detail Fact",
                "category": "dataSet",
                "updated": "2019-08-27 07:19:45",
                "contributor": "/gdc/account/profile/9bf17dfc423644e9bf92925bab65a3e9"
            }
        }
    }
]
