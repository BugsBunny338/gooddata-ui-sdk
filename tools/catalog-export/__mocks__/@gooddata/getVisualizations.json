[
    {
        "link": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/962",
        "locked": 0,
        "author": "/gdc/account/profile/8d381a33fff78cf82392c837246d362e",
        "tags": "",
        "created": "2018-04-27 14:23:11",
        "deprecated": "0",
        "identifier": "abhJpedgcfU2",
        "summary": "",
        "isProduction": 1,
        "title": "Labor Costs vs Scheduled Costs",
        "category": "visualizationObject",
        "updated": "2018-05-15 16:18:26",
        "unlisted": 0,
        "contributor": "/gdc/account/profile/9bf17dfc423644e9bf92925bab65a3e9"
    },
    {
        "link": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/964",
        "locked": 0,
        "author": "/gdc/account/profile/8d381a33fff78cf82392c837246d362e",
        "tags": "",
        "created": "2018-04-27 14:23:11",
        "deprecated": "0",
        "identifier": "acPWDTL2bJeX",
        "summary": "",
        "isProduction": 1,
        "title": "Total Sales by City",
        "category": "visualizationObject",
        "updated": "2018-05-15 16:18:26",
        "unlisted": 0,
        "contributor": "/gdc/account/profile/9bf17dfc423644e9bf92925bab65a3e9"
    },
    {
        "link": "/gdc/md/xxoajsqvo85zw3t786qljpxd0xkiena2/obj/965",
        "locked": 0,
        "author": "/gdc/account/profile/8d381a33fff78cf82392c837246d362e",
        "tags": "",
        "created": "2018-04-27 14:23:11",
        "deprecated": "0",
        "identifier": "aaBN5UG3dXu4",
        "summary": "",
        "isProduction": 1,
        "title": "$ Total Sales by Location",
        "category": "visualizationObject",
        "updated": "2018-05-15 16:18:26",
        "unlisted": 0,
        "contributor": "/gdc/account/profile/9bf17dfc423644e9bf92925bab65a3e9"
    }
]
