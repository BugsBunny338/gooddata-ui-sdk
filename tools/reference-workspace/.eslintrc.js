// (C) 2020 GoodData Corporation
module.exports = {
    extends: ["../../.eslintrc.js"],
    parserOptions: { tsconfigRootDir: __dirname, project: "./tsconfig.json" },
};
