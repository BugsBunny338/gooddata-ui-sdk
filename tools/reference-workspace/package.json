{
    "name": "@gooddata/reference-workspace",
    "version": "8.0.0-alpha.0",
    "author": "GoodData",
    "description": "GoodData SDK TypeScript skeleton",
    "repository": "https://github.com/gooddata/gooddata-ui-sdk/tree/master/libs/reference-workspace",
    "license": "BSD-3-Clause",
    "main": "dist/index.js",
    "typings": "dist/index.d.ts",
    "files": [
        "dist",
        "src",
        "LICENSE"
    ],
    "config": {
        "tslint": "./{src,test}/**/*.ts"
    },
    "scripts": {
        "dev": "tsc --watch",
        "copy-data": "cp -R src/recordings/* dist/recordings && rm -f dist/recordings/index.ts",
        "build": "rm -rf dist && tsc -p tsconfig.build.json && npm run copy-data",
        "test": "jest --watch",
        "test-once": "jest",
        "test-ci": "JEST_JUNIT_OUTPUT=./ci/results/test-results.xml jest --config jest.ci.json",
        "tslint": "tslint -p . -t verbose $npm_package_config_tslint",
        "tslint-ci": "mkdir -p ./ci/results && tslint -p . -t checkstyle -o ./ci/results/tslint-results.xml $npm_package_config_tslint",
        "prettier-check": "prettier --check '{src,test}/**/*.{ts,tsx,json,scss,md,yaml,html}'",
        "prettier-write": "prettier --write '{src,test}/**/*.{ts,tsx,json,scss,md,yaml,html}'",
        "validate": "tsc --noEmit && npm run tslint && npm run prettier-check",
        "validate-ci": "tsc --noEmit && npm run tslint-ci && npm run prettier-check"
    },
    "jest": {
        "preset": "ts-jest",
        "setupFilesAfterEnv": [
            "<rootDir>/jest.setup.ts"
        ],
        "testRegex": "(\\.(test|spec))\\.ts$",
        "testPathIgnorePatterns": [
            ",/node_modules/",
            "/dist/"
        ],
        "collectCoverageFrom": [
            "src/**/*.{ts}",
            "!**/*.d.ts"
        ],
        "collectCoverage": false,
        "coverageReporters": [
            "html"
        ],
        "globals": {
            "ts-jest": {
                "packageJson": "package.json"
            }
        }
    },
    "dependencies": {
        "@gooddata/sdk-backend-spi": "^8.0.0-alpha.0",
        "@gooddata/sdk-model": "^8.0.0-alpha.0",
        "lodash": "^4.17.15"
    },
    "devDependencies": {
        "@gooddata/catalog-export": "^8.0.0-alpha.0",
        "@gooddata/mock-handling": "1.0.0",
        "@gooddata/tslint-config": "1.0.0",
        "@types/jest": "^24.0.6",
        "@types/lodash": "^4.14.123",
        "@types/node": "^10.14.18",
        "jest": "^24.7.1",
        "jest-junit": "^3.0.0",
        "prettier": "~1.18.2",
        "ts-jest": "^24.1.0",
        "tslint": "^5.14.0",
        "tslint-config-prettier": "^1.18.0",
        "typescript": "3.6.3"
    }
}
