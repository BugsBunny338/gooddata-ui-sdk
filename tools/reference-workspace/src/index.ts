// (C) 2019-2020 GoodData Corporation

import * as ReferenceLdm from "./ldm/full";
import * as ReferenceLdmExt from "./ldm/ext";
import * as ReferenceRecordings from "./recordings";
import { DataSamples as ReferenceData } from "./recordings/dataSample";

export { ReferenceLdm, ReferenceLdmExt, ReferenceRecordings, ReferenceData };
