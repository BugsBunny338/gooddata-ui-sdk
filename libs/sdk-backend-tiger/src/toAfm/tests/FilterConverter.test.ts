// (C) 2020 GoodData Corporation
import {
    convertAbsoluteDateFilter,
    convertRelativeDateFilter,
    convertVisualizationObjectFilter,
} from "../FilterConverter";
import { absoluteFilter, relativeFilter, visualizationObjectFilter } from "./InvalidInputs.fixture";
import {
    newAbsoluteDateFilter,
    DateGranularity,
    newRelativeDateFilter,
    newPositiveAttributeFilter,
    newNegativeAttributeFilter,
} from "@gooddata/sdk-model";
import { ReferenceLdmExt, ReferenceLdm } from "@gooddata/reference-workspace";

describe("tiger filter converter from model to AFM", () => {
    describe("convert absolute date filter", () => {
        const Scenarios: Array<[string, any]> = [
            ["absolute date filter without 'to' attribute", absoluteFilter.withoutTo],
            ["absolute date filter without 'from' attribute", absoluteFilter.withoutFrom],
            [
                "absolute date filter from model to AFM with all attributes",
                newAbsoluteDateFilter(ReferenceLdmExt.ClosedDataDatasetRef, "2019-08-06", "2019-08-12"),
            ],
        ];

        it.each(Scenarios)("should convert %s", (_desc, input) => {
            expect(convertAbsoluteDateFilter(input)).toMatchSnapshot();
        });
    });
    describe("convert relative date filter", () => {
        const Scenarios: Array<[string, any]> = [
            [
                "year granularity",
                newRelativeDateFilter(ReferenceLdmExt.ClosedDataDatasetRef, DateGranularity.year, 2, 7),
            ],
            [
                "month granularity",
                newRelativeDateFilter(ReferenceLdmExt.ClosedDataDatasetRef, DateGranularity.month, -3, 9),
            ],
            [
                "quarter granularity",
                newRelativeDateFilter(ReferenceLdmExt.ClosedDataDatasetRef, DateGranularity.quarter, 25, -2),
            ],
            [
                "date granularity",
                newRelativeDateFilter(ReferenceLdmExt.ClosedDataDatasetRef, DateGranularity.date, -11, 0),
            ],
            [
                "week granularity",
                newRelativeDateFilter(ReferenceLdmExt.ClosedDataDatasetRef, DateGranularity.week, 50, 100),
            ],
            ["missing 'to' parameter", relativeFilter.withoutTo],
            ["mission 'from' parameter", relativeFilter.withoutFrom],
        ];
        it.each(Scenarios)("should return AFM relative date filter with %s", (_desc, input) => {
            expect(convertRelativeDateFilter(input)).toMatchSnapshot();
        });
    });
    describe("convert visualization object filter", () => {
        const Scenarios: Array<[string, any]> = [
            ["positive attribute filter", newPositiveAttributeFilter(ReferenceLdm.Product.Name, ["value"])],
            [
                "negative attribute filter",
                newNegativeAttributeFilter(ReferenceLdm.Product.Name, ["other value"]),
            ],
            ["null when filter is empty", newNegativeAttributeFilter(ReferenceLdm.Product.Name, [])],
            [
                "absolute date filter",
                newAbsoluteDateFilter(ReferenceLdmExt.ClosedDataDatasetRef, "2019-08-06", "2019-08-12"),
            ],
            [
                "relative date filter",
                newRelativeDateFilter(ReferenceLdmExt.ClosedDataDatasetRef, DateGranularity.date, 20, 30),
            ],
        ];
        it.each(Scenarios)("should return %s", (_desc, input) => {
            expect(convertVisualizationObjectFilter(input)).toMatchSnapshot();
        });

        it("should throw an error since tiger database only supports specifying attribute elements by value", () => {
            expect(() =>
                convertVisualizationObjectFilter(visualizationObjectFilter.positiveAttributeFilter),
            ).toThrowErrorMatchingSnapshot();
        });

        it("should throw an error since tiger database only supports specifying attribute elements by value", () => {
            expect(() =>
                convertVisualizationObjectFilter(visualizationObjectFilter.negativeAttributeFilter),
            ).toThrowErrorMatchingSnapshot();
        });

        it("should throw an error when visualization object filter does not support measure value filters", () => {
            expect(() =>
                convertVisualizationObjectFilter(visualizationObjectFilter.measureValueFilter),
            ).toThrowErrorMatchingSnapshot();
        });

        it("should filter out empty attribute filters and not cause RAIL-2083", () => {
            const emptyPositiveFilter = newPositiveAttributeFilter(ReferenceLdm.Product.Name, []);
            const emptyNegativeFilter = newNegativeAttributeFilter(ReferenceLdm.Product.Name, []);

            expect(convertVisualizationObjectFilter(emptyPositiveFilter)).toBeNull();
            expect(convertVisualizationObjectFilter(emptyNegativeFilter)).toBeNull();
        });
    });
});
