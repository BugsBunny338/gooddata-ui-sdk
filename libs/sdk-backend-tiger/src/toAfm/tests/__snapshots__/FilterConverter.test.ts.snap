// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tiger filter converter from model to AFM convert absolute date filter should convert absolute date filter from model to AFM with all attributes 1`] = `
Object {
  "absoluteDateFilter": Object {
    "dataSet": Object {
      "identifier": Object {
        "id": "closed",
        "type": "dataSet",
      },
    },
    "from": "2019-08-06",
    "to": "2019-08-12",
  },
}
`;

exports[`tiger filter converter from model to AFM convert absolute date filter should convert absolute date filter without 'from' attribute 1`] = `null`;

exports[`tiger filter converter from model to AFM convert absolute date filter should convert absolute date filter without 'to' attribute 1`] = `null`;

exports[`tiger filter converter from model to AFM convert relative date filter should return AFM relative date filter with date granularity 1`] = `
Object {
  "relativeDateFilter": Object {
    "dataSet": Object {
      "identifier": Object {
        "id": "closed",
        "type": "dataSet",
      },
    },
    "from": -11,
    "granularity": "GDC.time.date",
    "to": 0,
  },
}
`;

exports[`tiger filter converter from model to AFM convert relative date filter should return AFM relative date filter with missing 'to' parameter 1`] = `
Object {
  "relativeDateFilter": Object {
    "dataSet": Object {
      "identifier": Object {
        "id": "closed",
        "type": "dataSet",
      },
    },
    "from": 5,
    "granularity": "GDC.time.date",
    "to": NaN,
  },
}
`;

exports[`tiger filter converter from model to AFM convert relative date filter should return AFM relative date filter with mission 'from' parameter 1`] = `null`;

exports[`tiger filter converter from model to AFM convert relative date filter should return AFM relative date filter with month granularity 1`] = `
Object {
  "relativeDateFilter": Object {
    "dataSet": Object {
      "identifier": Object {
        "id": "closed",
        "type": "dataSet",
      },
    },
    "from": -3,
    "granularity": "GDC.time.month",
    "to": 9,
  },
}
`;

exports[`tiger filter converter from model to AFM convert relative date filter should return AFM relative date filter with quarter granularity 1`] = `
Object {
  "relativeDateFilter": Object {
    "dataSet": Object {
      "identifier": Object {
        "id": "closed",
        "type": "dataSet",
      },
    },
    "from": 25,
    "granularity": "GDC.time.quarter",
    "to": -2,
  },
}
`;

exports[`tiger filter converter from model to AFM convert relative date filter should return AFM relative date filter with week granularity 1`] = `
Object {
  "relativeDateFilter": Object {
    "dataSet": Object {
      "identifier": Object {
        "id": "closed",
        "type": "dataSet",
      },
    },
    "from": 50,
    "granularity": "GDC.time.week_us",
    "to": 100,
  },
}
`;

exports[`tiger filter converter from model to AFM convert relative date filter should return AFM relative date filter with year granularity 1`] = `
Object {
  "relativeDateFilter": Object {
    "dataSet": Object {
      "identifier": Object {
        "id": "closed",
        "type": "dataSet",
      },
    },
    "from": 2,
    "granularity": "GDC.time.year",
    "to": 7,
  },
}
`;

exports[`tiger filter converter from model to AFM convert visualization object filter should return absolute date filter 1`] = `
Object {
  "absoluteDateFilter": Object {
    "dataSet": Object {
      "identifier": Object {
        "id": "closed",
        "type": "dataSet",
      },
    },
    "from": "2019-08-06",
    "to": "2019-08-12",
  },
}
`;

exports[`tiger filter converter from model to AFM convert visualization object filter should return negative attribute filter 1`] = `
Object {
  "negativeAttributeFilter": Object {
    "displayForm": Object {
      "identifier": Object {
        "id": "label.product.id.name",
        "type": "label",
      },
    },
    "notIn": Object {
      "values": Array [
        "other value",
      ],
    },
  },
}
`;

exports[`tiger filter converter from model to AFM convert visualization object filter should return null when filter is empty 1`] = `null`;

exports[`tiger filter converter from model to AFM convert visualization object filter should return positive attribute filter 1`] = `
Object {
  "positiveAttributeFilter": Object {
    "displayForm": Object {
      "identifier": Object {
        "id": "label.product.id.name",
        "type": "label",
      },
    },
    "in": Object {
      "values": Array [
        "value",
      ],
    },
  },
}
`;

exports[`tiger filter converter from model to AFM convert visualization object filter should return relative date filter 1`] = `
Object {
  "relativeDateFilter": Object {
    "dataSet": Object {
      "identifier": Object {
        "id": "closed",
        "type": "dataSet",
      },
    },
    "from": 20,
    "granularity": "GDC.time.date",
    "to": 30,
  },
}
`;

exports[`tiger filter converter from model to AFM convert visualization object filter should throw an error since tiger database only supports specifying attribute elements by value 1`] = `"Tiger backend only allows specifying attribute elements by value"`;

exports[`tiger filter converter from model to AFM convert visualization object filter should throw an error since tiger database only supports specifying attribute elements by value 2`] = `"Tiger backend only allows specifying attribute elements by value"`;

exports[`tiger filter converter from model to AFM convert visualization object filter should throw an error when visualization object filter does not support measure value filters 1`] = `"Tiger backend does not support measure value filters"`;
